gui/src/actions.js

```diff
@@ -24,187 +27,109 @@ const receiveJobsSuccess = (json) => {
 export const RECEIVE_JOB_SPEC_RUN_SUCCESS = 'RECEIVE_JOB_SPEC_RUN_SUCCESS'
 export const RECEIVE_JOB_SPEC_RUN_ERROR = 'RECEIVE_JOB_SPEC_RUN_ERROR'
 
-const requestJobSpecRun = () => ({ type: REQUEST_JOB_SPEC_RUN })
-const receiveJobSpecRunSuccess = (json) => {
-  return {
+fetchActions.jobSpecRun = {
+  request: requestAction(REQUEST_JOB_SPEC_RUN),
+  receiveSuccess: json => ({
     type: RECEIVE_JOB_SPEC_RUN_SUCCESS,
     item: json.data.attributes
-  }
-}
-const receiveJobSpecRunNetworkError = (error) => {
-  return {
-    type: RECEIVE_JOB_SPEC_RUN_ERROR,
-    error: error,
-    networkError: true
-  }
-}
-export const fetchJobSpecRun = (id) => {
-  return dispatch => {
-    dispatch(requestJobSpecRun())
-    return getJobSpecRun(id)
-      .then(json => dispatch(receiveJobSpecRunSuccess(json)))
-      .catch(error => dispatch(receiveJobSpecRunNetworkError(error)))
-  }
+  }),
+  receiveNetworkError: error => requestNetworkError(RECEIVE_JOB_SPEC_RUN_ERROR, error)
 }
 
 export const REQUEST_CONFIGURATION = 'REQUEST_CONFIGURATION'
 export const RECEIVE_CONFIGURATION_SUCCESS = 'RECEIVE_CONFIGURATION_SUCCESS'
 export const RECEIVE_CONFIGURATION_ERROR = 'RECEIVE_CONFIGURATION_ERROR'
 
-const requestConfiguration = () => ({ type: REQUEST_CONFIGURATION })
-const receiveConfiguration = (json) => {
-  return {
+fetchActions.configuration = {
+  request: requestAction(REQUEST_CONFIGURATION),
+  receiveSuccess: json => ({
     type: RECEIVE_CONFIGURATION_SUCCESS,
     config: json.data.attributes
-  }
-}
-const receiveConfigurationNetworkError = (error) => {
-  return {
-    type: RECEIVE_CONFIGURATION_ERROR,
-    error: error,
-    networkError: true
-  }
-}
-export const fetchConfiguration = () => {
-  return dispatch => {
-    dispatch(requestConfiguration())
-    return getConfiguration()
-      .then(json => dispatch(receiveConfiguration(json)))
-      .catch(error => dispatch(receiveConfigurationNetworkError(error)))
-  }
+  }),
+  receiveNetworkError: error => requestNetworkError(RECEIVE_CONFIGURATION_ERROR, error)
 }
 
 export const REQUEST_BRIDGES = 'REQUEST_BRIDGES'
 export const RECEIVE_BRIDGES_SUCCESS = 'RECEIVE_BRIDGES_SUCCESS'
 export const RECEIVE_BRIDGES_ERROR = 'RECEIVE_BRIDGES_ERROR'
 
-const requestBridges = () => ({ type: REQUEST_BRIDGES })
-const receiveBridgesSuccess = (json) => {
-  return {
+fetchActions.bridges = {
+  request: requestAction(REQUEST_BRIDGES),
+  receiveSuccess: json => ({
     type: RECEIVE_BRIDGES_SUCCESS,
     count: json.meta.count,
     items: json.data.map(b => b.attributes)
-  }
-}
-const receiveBridgesNetworkError = (error) => {
-  return {
-    type: RECEIVE_BRIDGES_ERROR,
-    error: error,
-    networkError: true
-  }
+  }),
+  receiveNetworkError: error => requestNetworkError(RECEIVE_BRIDGES_ERROR, error)
 }
 
-export const fetchBridges = (page, size) => {
+function sendFetchActions (type) {
+  const getArgs = Array.prototype.slice.call(arguments, 1)
```

Comment:
Let's use a variadic function parameter to capture `getArgs` instead of `Array.prototype.slice.call(arguments, 1)`. Here's an example from our solidity deployer:

```js
    perform: async function perform (filename, ...contractArgs) {
      const compiled = compile(filename)
      const encodedArgs = encodeArgs(contractArgs, compiled.abi)
```

Reply:
@dimroc @j16r this takes the arguments from index 1 onwards. Which is everything excluding `type`. I don't think the spread operator supports that.